src/Microsoft.AspNet.OData.Shared/Formatter/Deserialization/ODataResourceSetWrapperBase.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Microsoft.OData;
+
+namespace Microsoft.AspNet.OData.Formatter.Deserialization
+{
+    /// <summary>
+    /// Encapsulates an <see cref="ODataResourceSet"/> and the <see cref="ODataResource"/>'s that are part of it.
+    /// </summary>
+    public abstract class ODataResourceSetWrapperBase : ODataItemBase
+    {
+        /// <summary>
+        /// /
+        /// </summary>
+        public ResourceSetType ResourceSetType;
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="ODataResourceSetWrapper"/>.
+        /// </summary>
+        /// <param name="item">The wrapped item.</param>
+        public ODataResourceSetWrapperBase(ODataResourceSetBase item)
+            : base(item)
+        {
+            ResourceItems = new List<ODataItemBase>();            
```

Comment:
>ODataItemBase [](start = 37, length = 13)

Can any type that derives from ODataItemBase go in an ODataResourceSetWrapperBase?  What about, for example, a NestedResourceInfoWrapper?

You may have to introduce an ODataResourceSetItemBase : ODataItemBase, and then have everything that currently derives from ODataItemBase AND can go in an ODataResourceSetWrapperBase (i.e., ODataResourceWrapperBase, ODataResourceSetWrapperBase, ODataDeltaLinkWrapper,..) inherit from that new interface.